test/integration/roles/test_ecs_ecr/tasks/main.yml

```diff
@@ -5,6 +5,10 @@
 - block:
     - name: When creating with check mode
       ecs_ecr: name='{{ ecr_name }}' region='{{ ec2_region }}'
+      environment:
```

Comment:
my only fear here would be if the variables got accidentally inherited from the environment and then used someone's production account details.